ext/sqlite3/sqlite3.c

```diff
@@ -2271,11 +2288,13 @@ static void php_sqlite3_result_object_free_storage(zend_object *object) /* {{{ *
 	sqlite3result_clear_column_names_cache(intern);
 
 	if (!Z_ISNULL(intern->stmt_obj_zval)) {
+		zval_ptr_dtor(&intern->stmt_obj_zval);
 		if (intern->stmt_obj && intern->stmt_obj->initialised) {
-			sqlite3_reset(intern->stmt_obj->stmt);
+			if (GC_REFCOUNT(&intern->stmt_obj->zo) == 0) {
+				sqlite3_reset(intern->stmt_obj->stmt);
```

Comment:
This is the bit I'm unsure about - to me it makes sense logically that we are only safe to reset the statement if the refcount on it is 0, but grepping the rest of the codebase I don't see much precedent for checking refcounts from extensions.

But, that said, the statement object is already inside free_list so we can be reasonably assured that it will be reset at some later point - so can we just remove this? if I remove this entire `if (intern->stmt_obj` block, I don't get any complaints from a debug build about detected memory leaks.